### PR TITLE
feat: Script for verifying zkSync contract configuration

### PR DESCRIPTION
**Pull request by pxrl**
This script implements some simple checks to ensure that the recently deployed zkSync contracts were configured correctly.

The script checks both the SpokePool and adapter contracts, which reside on the SpokePool chain and HubPool chain respectively. For production, this is 324 and 1.

### CODE REVIEW

File: scripts/verifyZkSyncDeployments.ts [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import { ContractFactory } from "ethers";
+import { Provider as zkProvider, utils as zkUtils } from "zksync-web3";
+import { ethers, toBN, findArtifactFromPath } from "../utils/utils";
+
+const textPadding = 30;
+
+/**
+ * l2RefundAddress and spokePoolDeploymentBlocks can change per deployment.
+ */
+const l2RefundAddress = "0x9A8f92a830A5cB89a3816e3D267CB7791c16b04D";
```
Comment (nicholaspai) [inline scripts/verifyZkSyncDeployments.ts:19]: This should not be the dev wallet as the mnemonic is shared too widely we should use one of the gckms keys

Comment (pxrl) [reply scripts/verifyZkSyncDeployments.ts]: Great catch - that was a mistake. Have fixed it now :+1:

**Comment by pxrl** [comment]
Also for reference, here's the output of the script at present:

```
$ for network in mainnet zksync; do yarn hardhat run --network ${network} scripts/ver >
yarn run v1.22.19
$ dev/across-protocol/contracts-v2/node_modules/.bin/hardhat run --network mainnet scripts/verifyZkSyncDeployments.ts
L1 WETH:                      0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
L2 refund address:            0x428AB2BA90Eba0a4Be7aF34C9Ac451ab061AC010
zkSyncMessageBridge:          0x32400084C286CF3E17e7B677ea9583e60a000324
zkERC20Bridge:                0x57891966931Eb4Bb6FB81430E6cE0A03AAbDe063
Done in 6.08s.
yarn run v1.22.19
$ dev/across-protocol/contracts-v2/node_modules/.bin/hardhat run --network zksync scripts/verifyZkSyncDeployments.ts
SpokePool chainId():          324
SpokePool getCurrentTime():   1691158933
SpokePool hubPool():          0xc186fA914353c44b2E33eBE05f21846F1048bEda
SpokePool crossDomainAdmin(): 0xc186fA914353c44b2E33eBE05f21846F1048bEda
Aliased admin address:        0xd297fA914353c44B2e33EBE05F21846f1048CFeB
SpokePool zkERC20Bridge():    0x11f943b2c77b743AB90f4A0Ae7d5A4e7FCA3E102
Done in 7.94s.
```

File: scripts/verifyZkSyncDeployments.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+const hubPools: { [chainId: number]: string } = {
+  324: "0xc186fA914353c44b2E33eBE05f21846F1048bEda",
+};
+
+const spokePools: { [chainId: number]: { adapter: string; spokePool: string } } = {
+  324: {
+    adapter: "0xE233009838CB898b50e0012a6E783FC9FeE447FB",
+    spokePool: "0xE0B015E54d54fc84a6cB9B666099c46adE9335FF",
+  },
+};
```
[the remark this answers is not in the document]
Comment (pxrl) [reply scripts/verifyZkSyncDeployments.ts]: Implemented: https://github.com/across-protocol/contracts-v2/pull/336/commits/b00406a4cf08266ec3c1a1fb8837aa1445c83642

File: scripts/verifyZkSyncDeployments.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+  const adapter = await adapterFactory.attach(adapterAddress);
+
+  const weth = tokens[chainId].find(({ symbol }) => symbol === "WETH")?.address;
+  const l1Weth = await adapter.l1Weth();
+  console.log("L1 WETH:".padEnd(textPadding) + l1Weth);
+  assert(l1Weth.toLowerCase() === weth?.toLowerCase(), `Unexpected L1 WETH: ${l1Weth} != ${weth}`);
+
+  const _l2RefundAddress = await adapter.l2RefundAddress();
+  console.log("L2 refund address:".padEnd(textPadding) + l2RefundAddress);
```
Comment (james-a-morris) [inline scripts/verifyZkSyncDeployments.ts:53]: This seems consistent enough to warrant making a helper fn

File: scripts/verifyZkSyncDeployments.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  324: [{ symbol: "WETH", address: "0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91" }],
+};
+
+async function verifyAdapter(
+  chainId: number,
+  signer: SignerWithAddress,
+  deployments: Record<string, any>
+): Promise<void> {
+  const contractName = "ZkSync_Adapter";
+  assert([1, 5].includes(chainId), `Unexpected chain ID (${chainId})`);
```
Comment (james-a-morris) [inline scripts/verifyZkSyncDeployments.ts:33]: Will we ever want more than just 1,5? We should probably place this in a top level variable

**Review by james-a-morris — COMMENTED** [review]
I think these verification scripts are well written. As a note: should we be adding more comments throughout the code?

**Comment by pxrl** [comment]
Dropping this because we now have [hubpool.ts](https://github.com/across-protocol/relayer-v2/blob/master/scripts/hubpool.ts) and [spokepool.ts](https://github.com/across-protocol/relayer-v2/blob/master/scripts/spokepool.ts) in `relayer-v2`.  If we need to verify configuration again I'll probably wrap those somehow instead.

**Comment by pxrl** [comment]
Dropping this because we now have [hubpool.ts](https://github.com/across-protocol/relayer-v2/blob/master/scripts/hubpool.ts) and [spokepool.ts](https://github.com/across-protocol/relayer-v2/blob/master/scripts/spokepool.ts) in `relayer-v2`.  If we need to verify configuration again I'll probably wrap those somehow instead.

**Pull request closed without merging**